### PR TITLE
feat(OverflowMenu): add light prop

### LINKED ISSUE #3934: Light Overflow Menu
### Summary

Light version of the Overflow menu component. It would be great to have `--light` modifier like in the Text input component.

Screenshot 2019-09-04 at 14 00 48

Development needed.

### Justification

Grey menu on grey background doesn't work well.

### PR DESCRIPTION
**Pull request by sabov**
Closes #3934 

Add `light` prop to `OverflowMenu` component for use on `$ui-01` container.

#### Changelog

**New**

- `light` prop in the `OverflowMenu` component
- `light` knob in the `OverflowMenu` storybook
- `light` prop test

**Changed**

- updated `DataTable` snapshots 

#### Testing / Reviewing

- run tests
- run react storybook
- activate the `light` knob in the `OverflowMenu` section

### CODE REVIEW

**Review by dakahn — COMMENTED** [review]
@snidersd Does the shadow around this component constitute a border in regards to WCAG 1.4.11?

In my testing the shadow meets 3:1 required contrast ratio. The reason I bring this up is if the our drop shadow is not a "border" from an accessibility standpoint then our existing failing contrast ratio of 1.1:1 will be made _worse_ by this change.

**Review by dakahn — CHANGES_REQUESTED** [review]
[WCAG 1.4.11](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html) requires a 3:1 contrast ratio for non-text interface elements and their adjacent colors. Common ways of meeting this requirement are 3:1 ratio contrast borders or improving the contrast between the interface element's background and it's adjacent colors (page background, parent background etc). 

As it stands many of our components do not yet meet this requirement as we rely primarily on a drop shadow or gradient effect to distinguish an interface element from it's adjacent colors. This technique _is permitted_ when the gradient used meets the 3:1 contrast ratio at it's darkest. Ours currently do not.

This situation is made worse in lighter themes by allowing an even lower contrast variant background color for these interface components.

**Review by dakahn — APPROVED** [review]
Okay! If `light` is only for dark themes. 👍

File: packages/components/src/components/overflow-menu/_overflow-menu.scss [hunk 1]
```diff
@@ -65,6 +65,12 @@
     box-shadow: 0 2px 6px 0 rgba(0, 0, 0, 0.3);
   }
 
+  .#{$prefix}--overflow-menu--light.#{$prefix}--overflow-menu--open,
+  .#{$prefix}--overflow-menu--light.#{$prefix}--overflow-menu--open
```
Comment (emyarod) [inline packages/components/src/components/overflow-menu/_overflow-menu.scss:69]: any reason for the duplicate selector?

File: packages/react/src/components/OverflowMenu/OverflowMenu-test.js [hunk 2]
```diff
@@ -80,6 +80,10 @@ describe('OverflowMenu', () => {
         0
       );
     });
+    it('should specify light version as expected', () => {
+      rootWrapper.setProps({ light: true });
+      expect(rootWrapper.props().light).toEqual(true);
```
Comment (emyarod) [inline packages/react/src/components/OverflowMenu/OverflowMenu-test.js:85]: should we also check that the `.bx--overflow-menu--light` class is applied?

File: packages/components/src/components/overflow-menu/_overflow-menu.scss (hunk 1 above)
Comment (sabov) [reply packages/components/src/components/overflow-menu/_overflow-menu.scss:69]: the second line doesn't have a comma so it goes with the third line

File: packages/react/src/components/OverflowMenu/OverflowMenu-test.js (hunk 2 above)
Comment (sabov) [reply packages/react/src/components/OverflowMenu/OverflowMenu-test.js:85]: Good point. I'll add it, thanks.

**Review by emyarod — APPROVED** [review]
looks good to me, thanks for adding those tests

**Comment by joshblack** [comment]
@laurenmrice do you have a sec to review? Is there another designer we should tag instead, if not?

**Review by laurenmrice — APPROVED** [review]
👍 light prop looks good to me. (Side note: I did notice that in the light prop for the dark themes the backgrounds for ui-02 are the same as ui-03 for the divider of the danger option, but this is a token value problem which i believe we have an issue for)

**Pull request merged**